src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax;
+
+import core.basesyntax.figure.*;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Random random = new Random();
+    private final ColorSupplier supplier = new ColorSupplier();
+    private static final int FIGURE_COUNT = 5;
+
+    public Figure getRandomFigure() {
+        Color color = supplier.getRandomColor();
+        return switch (random.nextInt(FIGURE_COUNT)) {
+            case 0 -> getRandomCircle(color);
+            case 1 -> getRandomRectangle(color);
+            case 2 -> getRandomIsoscelentTrapezoid(color);
```

Comment:
There is a typo in the method name `getRandomIsoscelentTrapezoid`. It should be `getRandomIsoscelesTrapezoid` to match the correct spelling of 'isosceles'.